server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/custom/authentication/strategy/ModularizeJmapRFC8621AuthenticationStrategyContract.java

```diff
@@ -41,58 +41,29 @@
-import org.junit.After;
-import org.junit.Before;
-import org.junit.Test;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
 
 import io.restassured.RestAssured;
 import io.restassured.http.Header;
 
 public abstract class ModularizeJmapRFC8621AuthenticationStrategyContract {
-    private AllowAuthenticationStrategy allowAuthenticationStrategy;
-    private DenyAuthenticationStrategy denyAuthenticationStrategy;
-    private GuiceJamesServer jmapServer;
-
-    protected abstract GuiceJamesServer createJmapServerWithAllowAuthenticationStrategy(AllowAuthenticationStrategy authenticationStrategy) throws IOException;
-
-    protected abstract GuiceJamesServer createJmapServerWithDenyAuthenticationStrategy(DenyAuthenticationStrategy authenticationStrategy) throws IOException;
+    public static Optional<List<String>> ALLOW_AUTHENTICATION_STRATEGY = Optional.of(List.of(AllowAuthenticationStrategy.class.getCanonicalName()));
+    public static Optional<List<String>> DENY_AUTHENTICATION_STRATEGY = Optional.of(List.of(DenyAuthenticationStrategy.class.getCanonicalName()));
+    public static Optional<List<String>> DEFAULT_STRATEGIES = Optional.empty();
 
-    protected abstract GuiceJamesServer createJmapServerWithDefaultAuthenticationStrategies() throws IOException;
-
-    protected abstract MailboxManager initMailboxManager();
+    private GuiceJamesServer jmapServer;
 
-    @Before
-    public void setup() {
-        MailboxManager mailboxManager = initMailboxManager();
-        allowAuthenticationStrategy = new AllowAuthenticationStrategy(mailboxManager);
-        denyAuthenticationStrategy = new DenyAuthenticationStrategy();
-    }
+    protected abstract GuiceJamesServer createJmapServer(GuiceJamesServer basedServer, Optional<List<String>> authOverride);
 
-    public void setupJamesServerWithCustomAuthenticationStrategy(Optional<AuthenticationStrategy> authenticationStrategy) throws Throwable {
-        jmapServer = authenticationStrategy.map(strategy -> {
-            if (strategy instanceof AllowAuthenticationStrategy) {
-                try {
-                    return createJmapServerWithAllowAuthenticationStrategy((AllowAuthenticationStrategy) strategy);
-                } catch (IOException e) {
-                    e.printStackTrace();
-                }
-            } else if (strategy instanceof DenyAuthenticationStrategy) {
-                try {
-                    return createJmapServerWithDenyAuthenticationStrategy((DenyAuthenticationStrategy) strategy);
-                } catch (IOException e) {
-                    e.printStackTrace();
-                }
-            }
-            return null;
-        }).orElse(createJmapServerWithDefaultAuthenticationStrategies());
+    public void setupJamesServerWithCustomAuthenticationStrategy(GuiceJamesServer basedServer, Optional<List<String>> authOverride) throws Throwable {
+        jmapServer = createJmapServer(basedServer, authOverride);
```

Comment:
Why create a new server if we already have one?

Reply:
I want to modify the server inside @RegisterExtension with custom auth for each test.